gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,396 @@
+            - Either quartiles or histogram of the data depending on plot_type of this metric.
+
+        Args:
+            data: 1D array of all values of the metric
+
+        Returns:
+            summary as a dict that can be serialized to JSON for storage.
+        """
+        if data.ndim != 1:
+            raise ValueError("Metric must be a 1D distribution to get summary.")
+        summary = {
+            "min": np.min(data).tolist(),
+            "max": np.max(data).tolist(),
+            "median": np.nanmedian(data).tolist(),
+            "mean": np.nanmean(data).tolist(),
+            "stddev": np.nanstd(data).tolist(),
+        }
+        if self._plot_type == self.PlotType.BOX:
+            summary.update({"quartiles": self._get_distribution_quartiles(data)})
+        elif self._plot_type == self.PlotType.HISTOGRAM:
+            summary.update({"histogram": self._get_distribution_histogram(data)})
+        return summary
+
+    def _get_distribution_quartiles(self, data: np.ndarray) -> Dict[int, float]:
+        """Computes quartiles for the provided 1D data distribution.
+
+        Metrics that are to be plotted as box plots contain quartiles in their summary.        
+
+        Args:
+            data: 1D distribution of metric values
+
+        Returns:
+            Quartiles of the data as a dict where keys are q0, q1, q2, q3, and q4
+        """
+        query = list(range(0, 101, 25))
+        quartiles = np.percentile(data, query)
+        output = {}
+        for i, q in enumerate(query):
+            output["q" + str(i)] = quartiles[i].tolist()
+        return output
+
+    def get_metric_as_dict(self) -> Dict[str, Any]:
+        """Provides a dictionary representation of the metric that can be serialized to JSON.
+
+        The dict contains a single element, for which the key is the name of the metric.
+        If metric is a distribution, the dict is in the below format: 
+        {
+            metric_name: {
+               DATA_KEY: [.. raw data if stored ..]
+               SUMMARY_KEY: {
+                    .. summary (stats) of distribution ..    
+               }
+            }
+        }
+        If the metric is scalar, it is stored simply as {metric_name: value}. 
+
+        Returns:
+            The metric as a dict representation explained above.
+        """
+        if self._dim == 0:
+            return {self._name: self._data.tolist()}
+
+        metric_dict = {SUMMARY_KEY: self.summary}
+        if self._data is not None:
+            metric_dict[DATA_KEY] = self._data.tolist()
+        return {self._name: metric_dict}
+
+    def save_to_json(self, json_filename: str) -> None:
+        """Saves this metric's dict representation to a JSON file.
+
+        Args:
+            Path to the json file.
+        """
+        io.save_json_file(json_filename, self.get_metric_as_dict())
+
+    @classmethod
+    def parse_from_dict(cls, metric_dict: Dict[str, Any]) -> GtsfmMetric:
```

Comment:
again i think clearer to just make this part of `parse_from_json`, if we will never call `parse_from_dict()` on its own anywhere. kind of confusing to figure out where this dict would come from. but `parse_from_json()` is very clear in terms of provenance

Reply:
GtsfmMetric.parse_from_dict is used in GtsfmMetricsGroup.parse_from_dict. 
Your point would be valid for GtsfmMetricsGroup.parse_from_dict. That I do not plan on using anywhere outside this class. But I felt both classes could have a similar structure. I could remove GtsfmMetricsGroup.parse_from_dict if you'd like that.

Reply:
ok, i dont feel too strongly about it